### PR TITLE
Implement customNode Setting for Each Crypto / Blockchain

### PR DESCRIPTION
**Pull request by kylanhurt**
The purpose of this task is to allow users to point the app at a **list** of custom nodes for each blockchain / cryptocurrency. Full functionality is pending implementation in the plugins.

Asana Task: https://app.asana.com/0/361770107085503/727899759988176/f

### CODE REVIEW

File: src/modules/UI/scenes/Settings/CurrencySettingsConnector.js [hunk 1]
```diff
@@ -4,17 +4,29 @@ import { connect } from 'react-redux'
...
+  isSetCustomNodesModalVisible: state.ui.scenes.settings.isSetCustomNodesModalVisible
 })
 const mapDispatchToProps = (dispatch: Dispatch, ownProps) => ({
   selectDenomination: denominationKey => {
     dispatch(setDenominationKeyRequest(ownProps.currencyCode, denominationKey))
-  }
+  },
+  toggleEnableCustomNodes: () => dispatch(toggleEnableCustomNodes(ownProps.currencyCode)),
+  // $FlowFixMe
+  toggleSetCustomNodesModalVisibility: (visibility: boolean | null) => dispatch(setCustomNodeModalVisibility(visibility)),
```
Comment (kylanhurt) [inline src/modules/UI/scenes/Settings/CurrencySettingsConnector.js]: Remove `null` as a possibility

File: src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  onChangeText = (input: string) => {
+    this.setState({
+      readableCustomNodesList: input
+    })
+  }
+
+  handleSave = () => {
+    if (this.state.readableCustomNodesList) {
```
Comment (kylanhurt) [inline src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: Empty array is considered truthy. Maybe use array.length?

Comment (kylanhurt) [reply src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: Also trim this.state.readableCustomNodesList of any extra spaces

File: src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          <Icon style={styles.txIDIcon} name={Constants.QUESTION_ICON} type={Constants.FONT_AWESOME} size={22} />
+        </InteractiveModal.Icon>
+
+        <InteractiveModal.Title>
+          <Text>{s.strings.settings_set_custom_nodes_modal_title}</Text>
+        </InteractiveModal.Title>
+
+        <InteractiveModal.Body>
+          <InteractiveModal.Description style={{textAlign: 'center'}}>{s.strings.settings_set_custom_nodes_modal_instructions}</InteractiveModal.Description>
+          <View style={styles.customNodesInputWrap}>
```
Comment (kylanhurt) [inline src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: Use `interactiveModal.item` instead of `<View>`

Comment (kylanhurt) [reply src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: I've decided to use my own `<View>` component to keep the styling simpler.

[the remark this answers is not in the document]
Comment (kylanhurt) [reply src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: Used my own `<View>` component here to keep the styling simple.

File: src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+              placeholder={s.strings.settings_set_custom_nodes_placeholder}
+              placeholderTextColor={rawStyle.placeholderText.color}
+              underlineColorAndroid={rawStyle.placeholderUnderline.color}
+              autoCorrect={false}
+            />
+          </View>
+        </InteractiveModal.Body>
+
+        <InteractiveModal.Footer>
+          <View style={styles.buttonsWrap}>
```
[the remark this answers is not in the document]
Comment (kylanhurt) [reply src/modules/UI/scenes/Settings/components/SetCustomNodesModal.ui.js]: Same here

**Comment by wolverineks** [comment]
when disabling custom nodes, the SetCustomModalNodesModal should not show

**Comment by wolverineks** [comment]
Cant turn custom nodes off

**Comment by wolverineks** [comment]
Too much spacing between denominations section and custom nodes section

**Comment by swansontec** [comment]
@kylanhurt As of the latest core, you now have `EdgeAccount.currencyTools.changePluginSettings`, so you can finally wire this up for real!

**Pull request closed without merging**